### PR TITLE
Add update_metrics to PairOrdinalRegressionTask

### PR DESCRIPTION
**Pull request by HaokunLiu**
This also clean up the _pair_sentence_forward function

### CODE REVIEW

**Comment by HaokunLiu** [comment]
#581

**Comment by sleepinyourhat** [comment]
@pruksmhc or @W4ngatang should review this—I'm not as familiar with the metrics code.

File: src/tasks/tasks.py [hunk 1]
```diff
@@ -359,9 +360,9 @@ def process_split(self, split, indexers) -> Iterable[Type[Instance]]:
         ''' Process split text into a list of AllenNLP Instances. '''
         return process_single_pair_task_split(split, indexers, is_pair=True,
                                               classification=False)
-    def update_metrics():
-        # currently don't support metrics for regression task
-        # TODO(Yada): support them! 
+    def update_metrics(self, logits, labels, tagmask=None):
+        self.scorer1(mean_squared_error(logits, labels))
```
Comment (pruksmhc) [inline src/tasks/tasks.py]: Add a comment here that this is average MSE (which is why you add a MSE here)

Comment (HaokunLiu) [reply src/tasks/tasks.py]: Done. Actually there is also similar comment in the __init__ of this task.

**Comment by HaokunLiu** [comment]
Once more, this branch is ready.

**Pull request merged**